CMakeLists.txt

```
@@ -856,13 +856,16 @@ if(CURL_DISABLE_LDAP)
 endif()
 
 # Check for idn2
-option(USE_LIBIDN2 "Use libidn2 for IDN support" ON)
+option(USE_LIBIDN2 "Use libidn2 for IDN support" OFF)
 if(USE_LIBIDN2)
   check_library_exists("idn2" "idn2_lookup_ul" "" HAVE_LIBIDN2)
   if(HAVE_LIBIDN2)
     set(CURL_LIBS "idn2;${CURL_LIBS}")
     check_include_file_concat("idn2.h" HAVE_IDN2_H)
   endif()
+  if(NOT HAVE_LIBIDN2 OR NOT HAVE_IDN2_H)
+    message(FATAL_ERROR "libidn2 not found")
```

```suggestion
    message(STATUC "libidn2 requested, but not found")
    set(USE_LIBIDN2 OFF)
```
What about just resetting `USE_LIBIDN2` to `OFF` and showing a warning instead?
This way we could leave it on by default and let it be used automatically e.g. on systems where it's present. Also retaining compatiblity with existing builds.